prometheus/src/main/scala/filodb/prometheus/query/PrometheusModel.scala

```diff
@@ -77,7 +76,8 @@ object PrometheusModel {
   }
 
   def toPromSuccessResponse(qr: filodb.query.QueryResult, verbose: Boolean): SuccessResponse = {
-    SuccessResponse(Data(toPromResultType(qr.resultType), qr.result.map(toPromResult(_, verbose))))
+    SuccessResponse(Data(toPromResultType(qr.resultType),
+      qr.result.map(toPromResult(_, verbose)).filter(_.values.size > 0)))
```

Comment:
Can we filter earlier in the transformation pipeline as follows:

```
qr.result.filter(_.rows.nonEmpty).map(toPromResult(_, verbose)))
```

Reply:
Ignore earlier comment. We would never have empty results. It would have NaNs.

That said, it is better to use `_.values.nonEmpty` instead of  `_.values.size > 0`